includes/rest-api/class-sensei-rest-api-course-utils-controller.php

```diff
@@ -93,8 +97,25 @@ public function register_routes() {
 	 * @return WP_REST_Response The json response.
 	 */
 	public function update_teacher( WP_REST_Request $request ) {
-		$post_id = $request->get_param( 'post_id' );
-		$teacher = $request->get_param( 'teacher' );
+		$post_id             = $request->get_param( 'post_id' );
+		$teacher             = $request->get_param( 'teacher' );
+		$module_custom_slugs = $request->get_param( 'custom_slugs' );
+
+		// If a custom slug is of a module that belongs to another teacher from another course, don't process farther.
+		if ( isset( $module_custom_slugs ) ) {
+			$module_custom_slugs = json_decode( sanitize_text_field( wp_unslash( $module_custom_slugs ) ) );
+			foreach ( $module_custom_slugs as $module_custom_slug ) {
+				$course_name = Sensei()->teacher::is_module_in_use_by_different_course_and_teacher( $module_custom_slug, $post_id, $teacher );
+				if ( $course_name ) {
+					return new WP_REST_Response(
+						[
+							'status' => 'module in use by different course and teacher',
+						],
+						WP_HTTP::OK
```

Comment:
Hmmm if I understand correctly, it means that in that case the user is going to try to update the teacher and although there is going to be a success response the update will not happen. Maybe it makes sense to return an error instead?

Reply:
Fair enough, not sure what status code to go with here, maybe `WP_HTTP::BAD_REQUEST`?